src/lib/CollisionPrevention/CollisionPrevention.cpp

```diff
@@ -216,52 +212,26 @@ void CollisionPrevention::_calculateConstrainedSetpoint(Vector2f &setpoint,
 						angle += math::radians(obstacle.angle_offset);
 					}
 
-					//split current setpoint into parallel and orthogonal components with respect to the current bin direction
 					Vector2f bin_direction = {cos(angle), sin(angle)};
-					Vector2f orth_direction = {-bin_direction(1), bin_direction(0)};
-					float sp_parallel = setpoint_sqrd.dot(bin_direction);
-					float sp_orth = setpoint_sqrd.dot(orth_direction);
 					float curr_vel_parallel = math::max(0.f, curr_vel.dot(bin_direction));
 
 					//calculate max allowed velocity with a P-controller (same gain as in the position controller)
 					float delay_distance = curr_vel_parallel * _param_mpc_col_prev_dly.get();
 					float vel_max_posctrl = math::max(0.f,
 									  _param_mpc_xy_p.get() * (distance - _param_mpc_col_prev_d.get() - delay_distance));
-					float vel_max_sqrd = vel_max_posctrl * vel_max_posctrl;
-
-					//limit the setpoint to respect vel_max by subtracting from the parallel component
-					if (sp_parallel > vel_max_sqrd) {
-						Vector2f setpoint_temp = setpoint_sqrd - (sp_parallel - vel_max_sqrd) * bin_direction;
-						float setpoint_temp_length = setpoint_temp.norm();
-
-						//limit sliding angle
-						float angle_diff_temp_orig = acos(setpoint_temp.dot(setpoint) / (setpoint_temp_length * setpoint_length));
-						float angle_diff_temp_bin = acos(setpoint_temp.dot(bin_direction) / setpoint_temp_length);
-
-						if (angle_diff_temp_orig > max_slide_angle_rad && setpoint_temp_length > 0.001f) {
-							float angle_temp_bin_cropped = angle_diff_temp_bin - (angle_diff_temp_orig - max_slide_angle_rad);
-							float orth_len = vel_max_sqrd * tan(angle_temp_bin_cropped);
-
-							if (sp_orth > 0) {
-								setpoint_temp = vel_max_sqrd * bin_direction + orth_len * orth_direction;
-
-							} else {
-								setpoint_temp = vel_max_sqrd * bin_direction - orth_len * orth_direction;
-							}
-						}
-
-						setpoint_sqrd = setpoint_temp;
+					Vector2f  vel_max_vec = bin_direction * vel_max_posctrl;
+					float vel_max_bin = vel_max_vec.dot(setpoint_dir);
+					float vel_setpoint_bin = setpoint.dot(bin_direction);
+
+					//limit the velocity
+					//do not react to obstacles more than 45 degree off the given stick input cos(45deg) = 0.71
+					if (vel_setpoint_bin > 0.71f * setpoint_length && vel_max_bin >= 0) {
```

Comment:
Sounds like this check
```
if (setpoint_dir.dot(bin_direction) > cosf(_param_colprev_max_detection_angle.get())) {
    ...
}
```
would be sufficient.

And that it would save a few cycles to do this check earlier

Reply:
thanks! that sounds like a good idea. Did not particularly like the magic number either...

Reply:
👍